src/redteam/util.ts

```diff
@@ -15,7 +15,8 @@ export async function retryWithDeduplication<T>(
   operation: (currentItems: T[]) => Promise<T[]>,
   targetCount: number,
   maxConsecutiveRetries: number = 2,
-  dedupFn: (items: T[]) => T[] = (items) => Array.from(new Set(items)),
+  dedupFn: (items: T[]) => T[] = (items) =>
+    Array.from(new Set(items.map((item) => JSON.stringify(item)))).map((item) => JSON.parse(item)),
```

Comment:
out of curiosity, why is this change necessary?

Reply:
generatePrompts now returns an array of objects instead of an array of strings, so the items will never be the same since they are just pointers to objects now instead of strings, we need to convert the objects into strings to see if they are dupes.